sharktank/sharktank/examples/export_paged_llm_v1.py

```diff
@@ -121,26 +126,32 @@ def generate_batch_prefill(bs: int):
             "tokens": {1: sl_dim},
             "seq_lens": {},
             "seq_block_ids": {1: block_dim},
-            "cache_state": cache_state_dynamic_shapes,
+            "cs": cache_state_dynamic_shapes * llama_config.tensor_parallelism_size,
         }
 
         print(f"Exporting prefill_bs{bs}")
 
         @fxb.export_program(
             name=f"prefill_bs{bs}",
-            args=(tokens, seq_lens, seq_block_ids, cache_state),
+            args=(
+                tokens,
+                seq_lens,
+                seq_block_ids,
+                [shard._data for shard in cache_state[0].shards],
+            ),
             dynamic_shapes=dynamic_shapes,
             strict=args.strict,
         )
-        def _(model, tokens, seq_lens, seq_block_ids, cache_state):
+        def _(model, tokens, seq_lens, seq_block_ids, cs):
+            cs = [SplitPrimitiveTensor(ts=cs, shard_dim=cache_state[0].shard_dim)]
```

Comment:
Do we always what a sharded cache state?

Reply:
Reworked to only happen for when sharded. We should consider making this constructor a helper so that when shard count is 1 it just returns the tensor.